Sources/StreamChat/Workers/ChannelUpdater.swift

```diff
@@ -17,19 +17,27 @@ class ChannelUpdater<ExtraData: ExtraDataTypes>: Worker {
     func update(
         channelQuery: _ChannelQuery<ExtraData>,
         channelCreatedCallback: ((ChannelId) -> Void)? = nil,
-        completion: ((Error?) -> Void)? = nil
+        completion: ((Result<ChannelPayload<ExtraData>, Error>) -> Void)? = nil
     ) {
         apiClient.request(endpoint: .channel(query: channelQuery)) { (result) in
             do {
                 let payload = try result.get()
                 channelCreatedCallback?(payload.channel.cid)
-                self.database.write { (session) in
+                self.database.write { session in
+                    try session.saveChannel(payload: payload)
+                    completion?(.success(payload))
+                }
+                self.database.write { session in
```

Comment:
why do we save channel twice? 🤔

Reply:
This seems wrong, nice catch 👍 Will fix it

Reply:
@evsaev Done ✅